### DISCUSSION TITLE
SET statement_timeout =3; not working with raw_sql set to false.

### DISCUSSION THREAD

**Discussion by sughosh-kaushik-eightfold** [sqlalchemy/sqlalchemy#9792] [Usage Questions]
### Describe the bug

We are using sqlalchemy to query redshift. We are trying to set statment_timeout = 1 and we are not able to do so with raw_sql=False. 
`SET statement_timeout = 5; query` the statement time is not coming into effect if we run the query on redshift.

### SQLAlchemy Version in Use

Flask-SQLAlchemy==2.4.3

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

Rdshift

### Python Version

3.11

### Operating system

Linux

### To Reproduce

```python
`SET statement_timeout = 5; query`
 execute above query with raw_sql as False
```

### Error

```
No Error 
```

**Comment by CaselIT** [comment]
Hi,

I don't know what `raw_sql=False` is. Also any reproduction code is missing, so moving to discussion

That said many drivers don't allow more than one statement in a single execute, so no `;`.
Run two executes, one for the set statement, the other for the actual query.

You may also use the before cursor execute event to set the timeout https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.ConnectionEvents.before_cursor_execute
